_drafts/practical_zephyr_05_west.md

```diff
@@ -0,0 +1,1246 @@
+    # --snip--
+    - name: dummy
+      revision: main
+      url: git@github.com:lmapii/practical-zephyr-t2-dummy.git
+      path: shared/dummy
+```
+
+In case you need to update such a shared dependency, make sure to push the changes to a new or existing branch, but **don't commit to the `manifest-rev` branch**. Also, after running `west update`, make sure to switch back to your working branch.
+
+This sounds brittle, but it really isn't. It isn't that easy to lose changes to files with a `west update` unless you commit these changes on your local `manifest-rev` branch and then run `west update`.
+
+Let's see this in action. In case you want to follow along, add your own dummy repository to the manifest - I'll be using the above project in my `west.yml` - a dummy repository that contains an empty `test.txt` file:
+
+```bash
+# after adding the "dummy" project to west.yml, update the project
+$ west update
+$ tree --dirsfirst -a -L 4 --filelimit 10 -I hal
+.
+├── .west
+│   └── config
+├── app
+├── deps
+│   ├── modules
+│   └── zephyr
+└── shared
+    └── dummy
+        └── test.txt
+```
+
+Without checking out a new branch, I can modify the contents of `test.txt` and run `west update`. The changes won't be lost and `west update` succeeds since the changes are not in conflict with any incoming change:
+
+```bash
+$ echo 123-test >> shared/dummy/test.txt
+$ west update
+...
+=== updating dummy (shared/dummy):
+--- dummy: fetching, need revision main
+From github.com:lmapii/practical-zephyr-t2-dummy
+ * branch            main       -> FETCH_HEAD
+M       test.txt
+HEAD is now at c8deb58 initial commit
+...
+
+$ cat shared/dummy/test.txt
+123-test
+```
+
+The output of `west update` shows me that I have in fact local changes to `test.txt`, but it doesn't reset the file. The changes are not lost.
+
+Since we're using `main` as `revision` for this dependency (e.g., since we're still very much in development), it _could_ happen that someone else is writing to `test.txt`, e.g., the value `not-a-test.` (let's not discuss using development branches etc., I'm just showing _West's_ behavior). Trying to run `west update` with an incoming change leads to an error - the local changes are still not lost:
+
+```bash
+$ west update
+...
+=== updating dummy (shared/dummy):
+--- dummy: fetching, need revision main
+remote: Enumerating objects: 5, done.
+remote: Counting objects: 100% (5/5), done.
+remote: Total 3 (delta 0), reused 3 (delta 0), pack-reused 0
+Unpacking objects: 100% (3/3), 229 bytes | 57.00 KiB/s, done.
+From github.com:lmapii/practical-zephyr-t2-dummy
+ * branch            main       -> FETCH_HEAD
+error: Your local changes to the following files would be overwritten by checkout:
+        test.txt
+Please commit your changes or stash them before you switch branches.
+Aborting
+...
+ERROR: update failed for project dummy
+```
+
+The only irreversible mistake that we can make is to commit changes to the `manifest-rev` branch. Let's do just that:
```

Comment:
From the example here it looks like the commit is made from a detached head, not from the `manifest-rev`

Reply:
Yes, but that's the actual output of the command line. That was actually the point I was trying to make later in the article: The Git output of West can be very misleading.